### PR TITLE
Feature/gramine

### PR DESCRIPTION
**Pull request by james-toussaint**

### CODE REVIEW

**Review by mcornaton — APPROVED** [review]

File: src/test/java/com/iexec/common/tee/TeeUtilsTests.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    })
+    void isTeeTag(int tag) {
+        Assertions.assertTrue(TeeUtils.isTeeTag(toByte32HexString(tag)));
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {
+            0b0000, // 0x0: empty
+            0b0001, // 0x1: missing TEE runtime framework
+            0b1001, // 0x7: not third TEE runtime framework ({Scone, Gramine, ?})
```
Comment (jbern0rd) [inline src/test/java/com/iexec/common/tee/TeeUtilsTests.java]: This value is `0x9`, not `0x7`

File: gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-version=6.0.1
+version=gramine
```
Comment (jbern0rd) [inline gradle.properties]: do not forget to revert this change :pray:

File: src/test/java/com/iexec/common/tee/TeeUtilsTests.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    @Test
+    void hasNotTeeGramineInTag() {
+        Assertions.assertFalse(TeeUtils.hasTeeGramineInTag(toByte32HexString(0b0011))); // 0x3: Scone
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {
+            0b0101, // 0xa: exact match
+            0b11110101, // 0xfa: contains
```
Comment (jbern0rd) [inline src/test/java/com/iexec/common/tee/TeeUtilsTests.java]: This is `0xf5`, `0xfa` would be `0b11111010`

File: src/test/java/com/iexec/common/tee/TeeUtilsTests.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    })
+    void hasTeeRuntimeFrameworkBitsInTag(int tag) {
+        int expectedBits = 0b0101; //0xa
```
Comment (jbern0rd) [inline src/test/java/com/iexec/common/tee/TeeUtilsTests.java]: Same, this is `0x5`, not `0xa` which is a tag without tee

File: src/main/java/com/iexec/common/tee/TeeUtils.java [hunk 5]
```diff
@@ -18,21 +18,87 @@
...
+     * @return {@link TeeEnclaveProvider} matching given {@code hexTag}
+     * or {@literal null} if tag is not a TEE tag or if there is no match.
+     */
+    public static TeeEnclaveProvider getTeeEnclaveProvider(String hexTag) {
+        for (Map.Entry<Integer, TeeEnclaveProvider> teeEnclaveProvider : TEE_BITS_TO_ENCLAVE_PROVIDER.entrySet()) {
+            if (hasTeeRuntimeFrameworkBitsInTag(teeEnclaveProvider.getKey(), hexTag)) {
+                return teeEnclaveProvider.getValue();
+            }
+        }
+        return null;
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/tee/TeeUtils.java:99]:
```suggestion
        //TODO: Add TeeEnclaveProvider.UNDEFINED
        return null;
```
?

File: src/main/java/com/iexec/common/task/TaskDescription.java [hunk 6]
```diff
@@ -113,6 +116,14 @@ public boolean containsPostCompute() {
                 !StringUtils.isEmpty(teePostComputeFingerprint);
     }
 
+    public String getAppCommand() {
+        String appArgs = appEnclaveConfiguration.getEntrypoint();
+        if (!StringUtils.isEmpty(cmd)) {
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/task/TaskDescription.java:122]:
```suggestion
        //TODO: Add unit test
        if (!StringUtils.isEmpty(cmd)) {
```

File: gradle.properties (hunk 2 above)
Comment (james-toussaint) [inline gradle.properties]:
```suggestion
version=6.0.1
```

File: src/test/java/com/iexec/common/tee/TeeUtilsTests.java (hunk 4 above)
Comment (james-toussaint) [reply src/test/java/com/iexec/common/tee/TeeUtilsTests.java]: Thanks updated here https://github.com/iExecBlockchainComputing/iexec-common/pull/339/commits/87d67579e01eb56540203e12c93a0dd3aadf123b !

**Review by jbern0rd — APPROVED** [review]
